CIP-0006/CIP-0006.md

```diff
@@ -0,0 +1,391 @@
+---
+CIP: 0006
+Title: Stake Pool Extended Metadata
+Authors: Markus Gufler <gufmar@gmail.com>, Mike Fullman <mike@fullman.net>
+Comments-URI:
+Status: Draft
+Type: Standards
+Created: 2020-07-20 original draft
+Updated: 2020-11-24 2nd key-pair for validation
+Updated: 2021-02-08 json schema
+License: CC-BY-4.0
+---
+
+## Abstract
+
+This CIP defines the concept of extended metadata for pools that is referenced from the pool registration data stored on chain.
+
+## Motivation
+
+As the ecosystem around Cardano stake pools proliferate so will the desire to slice, organize and search pool information dynamically.  Currently the metadata referenced on chain provides 512 bytes that can be allocated across the four information categories ([delegation-design-specification Section 4.2)](https://hydra.iohk.io/build/790053/download/1/delegation_design_spec.pdf):
+
+| key           | Value                                |  Rules  |
+| ---           | ---                                  |  ---  |
+|  `ticker` | Pool ticker.  uppercase | 5 Characters Maximum, Uppercase letters and numbers |
+|  `description` | Pool Description.  Text that describes the pool | 50 Characters Maximum |
+|  `homepage` | A website URL for the pool  | 64 Characters Maximum, must be a valid URL |
```

Comment:
Is this really accurate? As far as I can tell, the [Design Specification (Section 4.2)](https://hydra.iohk.io/build/790053/download/1/delegation_design_spec.pd) doesn't impose an explicit character limit on the "description" and "homepage" metadata fields. All it requires is a total size of the on-chain metadata of 512 bytes or less.

Reply:
good point. I'm not sure what is effectively true, but I'm aware of multiple different origins and specifications. 

For example the incentivised testnet metadata registry
https://github.com/cardano-foundation/incentivized-testnet-stakepool-registry#submission-well-formedness-rules
afaik this was used as a template for the mainnet metadata registration file. 

I also remember another proposal/definition/example I can't find now. 
And there  is what currently is implemented by the SMASH server as input validation
https://github.com/input-output-hk/smash/blob/479fc6d8fa62537cd6c8560176d6a8a7ffda6e9e/smash-servant-types/src/Cardano/SMASH/Types.hs#L212-L252

With the additional fields, we need to increase the current max size from 512 to (proposed) 1024 bytes
By not specifying additional max sizes for individual fields would make it very unpredictable (for example for UI design)
One could consciously use the shortest possible URLs, and fill an unlimited description or name field with >900 characters. 

So I propose to align to what Smash server currently has implemented and is also using in his SQL DB schema.